src/snowflake/snowpark/functions.py

```diff
@@ -8758,7 +8758,7 @@ def udtf(
     else:
         udtf_registration_method = session.udtf.register
 
-    if handler is None:
+    if handler is None and "object_name" not in kwargs:
```

Comment:
should this be `... and kwargs.get("object_name") is None` to handle the None case inclusively?